src/org/exist/xquery/functions/fn/FunMatches.java

```diff
@@ -209,22 +210,32 @@ public NodeSet preSelect(Sequence contextSequence, boolean useContext) throws XP
         final int indexType = Optimize.getQNameIndexType(context, contextSequence, contextQName);
         if (LOG.isTraceEnabled())
             {LOG.trace("Using QName index on type " + Type.getTypeName(indexType));}
-		
-        String pattern;
-		
-		if( isCalledAs( "matches-regex" ) ) {
-			pattern = getArgument(1).eval(contextSequence).getStringValue();
-		} else {
-			pattern = translateRegexp(getArgument(1).eval(contextSequence).getStringValue());
-		}
-		
-        boolean caseSensitive = true;
-        int flags = 0;
+
+        final int flags;
         if(getSignature().getArgumentCount() == 3) {
             final String flagsArg = getArgument(2).eval(contextSequence).getStringValue();
-            caseSensitive = (flagsArg.indexOf('i') == Constants.STRING_NOT_FOUND);
             flags = parseFlags(flagsArg);
+        } else {
+            flags = 0;
```

Comment:
i'd probably implement this not as a 'final' variable, since it increases readability. my 2 cts